### PR TITLE
build: update to rules_nodejs 1.6.0

### PR DESCRIPTION
**Pull request by trik**

### CODE REVIEW

File: tools/defaults.bzl [hunk 1]
```diff
@@ -258,3 +259,36 @@ def ng_web_test_suite(deps = [], static_css = [], bootstrap = [], tags = [], **k
         tags = ["native"] + tags,
         **kwargs
     )
+
+def _devmode_js_sources_impl(ctx):
```
Comment (devversion) [inline tools/defaults.bzl:263]: Can we move this into a separate file? This is not a `default` so it should not be in `defaults.bzl`.

I'd propose `tools/es5-named-output.bzl` and the macro should be more specific to our case. Since we now need to copy this anyway, we should only use stuff we need (we need no manifest).

The rule should be named `es5_named_output` with a comment explaining what it does. Preferably we also link to https://github.com/bazelbuild/rules_nodejs/blob/a167311c025be2a77ba0d84e6a2ddcafe1c0564d/packages/typescript/src/internal/build_defs.bzl#L312 for context.

File: tools/postinstall/manifest_externs_hermeticity.patch [hunk 2]
```diff
@@ -1,11 +1,11 @@
 diff --git node_modules/@bazel/typescript/internal/build_defs.bzl node_modules/@bazel/typescript/internal/build_defs.bzl
-index 7fef44c..4e25ed7 100644
+index 03cb68fdf..2f604234c 100755
 --- node_modules/@bazel/typescript/internal/build_defs.bzl
 +++ node_modules/@bazel/typescript/internal/build_defs.bzl
-@@ -93,25 +93,23 @@ def _filter_ts_inputs(all_inputs):
-         if f.path.endswith(".js") or f.path.endswith(".ts") or f.path.endswith(".json")
+@@ -102,17 +102,17 @@ def _filter_ts_inputs(all_inputs):
+         if f.extension in ["js", "jsx", "ts", "tsx", "json"]
```
Comment (devversion) [inline tools/postinstall/manifest_externs_hermeticity.patch:6]: What's the reason for this change?

File: tools/defaults.bzl [hunk 3]
```diff
@@ -249,7 +250,7 @@ def ng_web_test_suite(deps = [], static_css = [], bootstrap = [], tags = [], **k
             # Note: when changing the browser names here, also update the "yarn test"
             # script to reflect the new browser names.
             "@io_bazel_rules_webtesting//browsers:chromium-local",
-            "@io_bazel_rules_webtesting//browsers:firefox-local",
+            # "@io_bazel_rules_webtesting//browsers:firefox-local",
```
Comment (devversion) [inline tools/defaults.bzl:253]: What's the reason for this?

File: tools/postinstall/manifest_externs_hermeticity.patch (hunk 2 above)
Comment (trik) [reply tools/postinstall/manifest_externs_hermeticity.patch:6]: no reason, probably I made something when creating the git diff patch

File: tools/defaults.bzl (hunk 3 above)
Comment (trik) [reply tools/defaults.bzl:253]: oops, i'll fix this one, was just for local testing, for some reason i can't get headless firefox to run properly on my mac

File: WORKSPACE [hunk 4]
```diff
@@ -8,8 +8,8 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 # Add NodeJS rules
 http_archive(
     name = "build_bazel_rules_nodejs",
-    sha256 = "d0c4bb8b902c1658f42eb5563809c70a06e46015d64057d25560b0eb4bdc9007",
-    urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/1.5.0/rules_nodejs-1.5.0.tar.gz"],
+    sha256 = "f9e7b9f42ae202cc2d2ce6d698ccb49a9f7f7ea572a78fd451696d03ef2ee116",
```
Comment (josephperrott) [inline WORKSPACE:11]: The `build_bazel_rules_nodejs` archive needs to be updated in `integration/bazel/WORKSPACE` as well.

Comment (josephperrott) [reply WORKSPACE:11]: Whoops, wrong repository.

File: tools/postinstall/manifest_externs_hermeticity.patch (hunk 2 above)
Comment (trik) [reply tools/postinstall/manifest_externs_hermeticity.patch:6]: sorry, i was checking this one, but the reason is that build_defs has changes in version 1.6.0 and the post install script failed to apply the patch (the 2nd hunk)
https://github.com/bazelbuild/rules_nodejs/blob/bd83907954d1b3cd53dff01c37e3c4964890ef5c/packages/typescript/src/internal/build_defs.bzl

Comment (devversion) [reply tools/postinstall/manifest_externs_hermeticity.patch:6]: I see. makes sense. I thought it was part of our patch, but it's obviously just needed an update of the base for getting the diff.

**Comment by jelbourn** [comment]
@josephperrott feel free to add "merge ready" for build stuff like this
